channels/tasks.py

```diff
@@ -376,3 +378,53 @@ def send_invitation_email(channel_invitation_id):
             )
         )
     )
+
+
+@app.task
+def maybe_repair_post_in_host_listing(channel_name, reddit_post_id):
+    """
+    Repair a post on the condition that it's in the new post listing but not in the hot listing
+
+    Args:
+        channel_name(str): the channel name to search for the post in
+        reddit_post_id(str): the reddit post id to check
+    """
+    admin_api = get_admin_api()
+
+    limit = settings.OPEN_DISCUSSIONS_HOT_POST_REPAIR_LIMIT
+
+    def _find_missing_submission():
+        """Find the missing submission"""
+        new_posts = admin_api.list_posts(channel_name, SORT_NEW_LISTING_PARAMS)
+        new_posts_by_id = {post.id: post for post in islice(new_posts, limit)}
+        new_posts_ids = set(new_posts_by_id.keys())
+
+        hot_posts = admin_api.list_posts(channel_name, SORT_HOT_LISTING_PARAMS)
+        hot_posts_ids = {post.id for post in islice(hot_posts, limit)}
+
+        missing_post_ids = new_posts_ids - hot_posts_ids
+
+        print(new_posts_ids)
+        print(hot_posts_ids)
+        print(missing_post_ids)
+
+        if reddit_post_id in missing_post_ids and reddit_post_id in new_posts_ids:
+            return new_posts_by_id[reddit_post_id]
+        return None
+
+    # check to see if this submission existing in hot listing
+    submission = _find_missing_submission()
+
+    if submission is None:
+        return
+
+    submission.upvote()
+    submission.clear_vote()
+
+    # check again and if it's missing still, just report an error so we can investigate
+    if _find_missing_submission() is not None:
+        log.error(
+            "Failed to repair submission %s missing from hot posts in channel %s",
+            reddit_post_id,
+            channel_name,
+        )
```

Comment:
so `_find_missing_submission` is going to look at the first page of `new` and `hot` submissions and will return the submission if a post with the id we're looking for is in `new` but not in `hot`. this makes sense to me, it tells us generally if we need to repair it or not. I think it's possible though that the post is on the second or third page of the `hot` results, so we might end up repairing a post in a situation where we dont absolutely need to, but that's ok I think.

I'm wondering though about here where we use `_find_missing_submission` again to check whether the repair was successful. In the case where the repair was successful but the post is on the second page of the `hot` listings, wouldn't `_find_missing_submission` still return a post object, causing us to raise this error incorrectly?

admittedly, right now we might not have enough traffic on any of our channels that this is a concern, but still, what do you think? I'm not sure there is another good way to exhaustively check that the repair was successful, without trawling through every page of the `hot` listing to exhaustively check if it's _somewhere_ in there.

Reply:
Yeah, it's possible that it may get pushed onto the second page, but it's unlikely. We'd probably need to be getting some combination of tens of posts per minute and hundreds of upvotes per minute. This fix definitely falls in the category of "stopgap" and that's also why I put in a configuration for how many posts back into the history to check because that will allow us to increase that (at the expense of site performance).

Reply:
makes sense, I think if we are ever dealing with lots of traffic like that we will hopefully already have migrated away from reddit anyhow...